administration/src/bp-modules/self-service/hooks/useCardGeneratorSelfService.tsx

```diff
@@ -35,13 +37,21 @@ type UseCardGeneratorSelfServiceReturn = {
 const useCardGeneratorSelfService = (): UseCardGeneratorSelfServiceReturn => {
   const projectConfig = useContext(ProjectConfigContext)
   const appToaster = useAppToaster()
-  const [selfServiceCard, setSelfServiceCard] = useState(
-    initializeCard(projectConfig.card, undefined, { expirationDate: null })
-  )
+  const [cardQueryParams, setSearchParams] = useSearchParams()
+  const [selfServiceCard, setSelfServiceCard] = useState(() => {
+    const headers = getHeaders(projectConfig)
+    const values = headers.map(header => cardQueryParams.get(header))
+    return initializeCardFromCSV(projectConfig.card, values, headers, undefined, true)
+  })
   const [isLoading, setIsLoading] = useState<boolean>(false)
   const [selfServiceState, setSelfServiceState] = useState<CardSelfServiceStep>(CardSelfServiceStep.form)
   const [deepLink, setDeepLink] = useState<string>('')
   const [code, setCode] = useState<CreateCardsResult | null>(null)
+
+  useEffect(() => {
+    setSearchParams(undefined, { replace: true })
```

Comment:
:wrench:
Could you perhaps move this inside the initializing method to avoid the extra use effect and possible side effects?

Reply:
i think its better to move this to the `createKoblenzPass` task to only remove them if the card was created successfully, so the params will not be removed if there is an page refresh

Reply:
@f1sh1918 pls check if that's what you mean

Reply:
i think you can just reuse the hook in `CardSelfServiceForm`
```js
const [_, setSearchParams] = useSearchParams()
```
and execute the function as you did. Then you don't have to pass down the function neither to `CardSelfServiceForm` nor adjust the `useCardSelfService` hook

Reply:
hopefully, it's good to go now, thx 🙈